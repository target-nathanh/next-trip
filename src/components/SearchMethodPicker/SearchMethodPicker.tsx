import React, { useEffect, useState } from 'react';

import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { SearchMode } from '../../common/constants';
import { useHistory, useLocation } from 'react-router-dom';

const SearchMethodPicker: React.FC = () => {
  const history = useHistory();
  const [selectedMethod, setSelectedMethod] = useState<string>(
    history.location.pathname !== '/' ? history.location.pathname.split('/')[1] : SearchMode.ROUTE
  );

  const location = useLocation();

  useEffect(() => {
    if (location.pathname !== '/') {
      setSelectedMethod(location.pathname.split('/')[1]);
    }
  }, [location]);

  const selectedMethodChanged = (method: string) => {
    setSelectedMethod(method);
    history.push(`/${method}`);
  };
  return (
    <div>
      <Tabs
        value={selectedMethod}
        onChange={(event: React.ChangeEvent<{}>, newValue: string) =>
          selectedMethodChanged(newValue)
        }
        indicatorColor="primary"
        textColor="primary"
        centered
      >
        <Tab label="By Route" value={SearchMode.ROUTE} />
        <Tab label="By Stop #" value={SearchMode.STOP} />
      </Tabs>
    </div>
  );
};

export default SearchMethodPicker;
