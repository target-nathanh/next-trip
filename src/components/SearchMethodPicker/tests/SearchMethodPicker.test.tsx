import React from 'react';
import { render, screen } from '@testing-library/react';
import SearchMethodPicker from '../SearchMethodPicker';
import { MemoryRouter } from 'react-router-dom';

describe('SearchMethodPicker', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchMethodPicker />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('selects Route as default option', () => {
    render(
      <MemoryRouter>
        <SearchMethodPicker />
      </MemoryRouter>
    );
    const routeOption = screen.getByText('By Route');
    expect(routeOption.parentElement?.getAttribute('aria-selected')).toEqual('true');

    const stopOption = screen.getByText('By Stop #');
    expect(stopOption.parentElement?.getAttribute('aria-selected')).toEqual('false');
  });

  test('selects route from url if available on load', () => {
    render(
      <MemoryRouter initialEntries={['/stop']}>
        <SearchMethodPicker />
      </MemoryRouter>
    );
    const routeOption = screen.getByText('By Route');
    expect(routeOption.parentElement?.getAttribute('aria-selected')).toEqual('false');

    const stopOption = screen.getByText('By Stop #');
    expect(stopOption.parentElement?.getAttribute('aria-selected')).toEqual('true');
  });
});
