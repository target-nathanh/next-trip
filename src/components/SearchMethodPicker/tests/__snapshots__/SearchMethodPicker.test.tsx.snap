// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMethodPicker matches snapshot 1`] = `
<div>
  <div
    class="MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed"
      style="overflow: hidden;"
    >
      <div
        class="MuiTabs-flexContainer MuiTabs-centered"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            By Route
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            By Stop #
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorPrimary-2 MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</div>
`;
